GoogleForCreators/web-stories-wp
# Templates: Show colors as saved story colors

## Context

When a template is selected from the dashboard it should load the colors of that template as the the saved story colors. 

## Summary

Adding the template colors to the story data that's generating a story so that we get the template's colors as saved story colors. 

## Relevant Technical Choices
- The story structure wants colors as rgb values while template metadata assigns hex values as colors. Converting template colors when a template is selected to be used as a story and passed into the story as `currentStoryStyles.colors` so that they load as the saved story colors. 
- Update `Swatch` `pattern` propType to be more descriptive 
- Added 2 `data-testid`s to get counts of colors for an e2e test. the color swatches on a detail template view and color groups used as swatches in the saved colors panel. 
- Update the `Template` e2e test to track how colors present for a given template (the 'Fresh and Bright', id: 1 template) and then checking the same colors are shown when the template loads as a story in the saved colors. Left all other checks in play. Only real difference is that the 'use template' button that's getting clicked is the one on the detail view. 


## To-do

- [x] update to use patters util instead of polished 
- [x]  Check colors not just length of list in e2e test

## User-facing changes

Now when you use a template you will get the template's colors in the saved story colors panel. 

![saved template colors](https://user-images.githubusercontent.com/10720454/131754046-9aa03c28-90d6-4f10-97ea-72c860774bb7.gif)


## Testing Instructions

- Go to dashboard templates, go to the detail view of a template. 
- Take note of the colors listed below the template description. 
- Now use that template as a story and go to the 'saved colors' panel. 
- The same colors that are in the template detail view should be listed as the saved story colors (not global). 




This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---



Fixes #8828 
Fixes #8829
